File: chapter01/src/main/java/com/example/demo/DaoFactory.java
package com.example.demo;

import com.example.demo.users.UserDao;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class DaoFactory {

    @Bean
    public DConnectionMaker connectionMaker() {
        return new DConnectionMaker();
    }

//   @Bean
//     public UserDao userDao() {
//         return new UserDao(connectionMaker());
//     }

//    public UserDao accountDao() {
//        return new AccountDao(connectionMaker());
//    }
//
//    public UserDao messageDao() {
//        return new MessageDao(connectionMaker());
//    }
}

File: chapter05/src/main/kotlin/com/example/demo/DaoFactory.kt
package com.example.demo

import com.example.demo.users.DummyMailSender
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.mail.MailSender
import org.springframework.mail.javamail.JavaMailSenderImpl

@Configuration
class DaoFactory {

    @Bean
    fun mailSender(): MailSender {
        val mailSender = DummyMailSender()
//        mailSender.host = "mail.server.com"

        return mailSender
    }
}